neon_core/util/skill_utils.py

```diff
@@ -23,29 +23,72 @@
 # NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-from os.path import expanduser
+import git
+import json
+import requests
+
+from os import listdir
+from tempfile import mkdtemp
+from shutil import rmtree
+from os.path import expanduser, join
+from ovos_skills_manager.skill_entry import SkillEntry
 from ovos_skills_manager.osm import OVOSSkillsManager
-from ovos_skills_manager.session import SESSION as requests, set_github_token, clear_github_token
+from ovos_skills_manager.session import SESSION, set_github_token, clear_github_token
+from ovos_skills_manager.github import normalize_github_url, get_branch_from_github_url
 from neon_utils.configuration_utils import get_neon_skills_config
 from neon_utils import LOG
 
 
+def get_neon_skills_data(skill_meta_repository: str = "https://github.com/neongeckocom/neon_skills",
+                         branch: str = None,
+                         repo_metadata_path: str = "skill_metadata") -> dict:
+    """
+    Get skill data from configured neon_skills repository.
+    :param skill_meta_repository: URL of skills repository containing metadata
+    :param branch: branch of repository to checkout
+    :param repo_metadata_path: Path to repo directory containing json metadata files
+    """
+    skills_data = dict()
+    temp_download_dir = mkdtemp()
+    git.Repo.clone_from(skill_meta_repository, temp_download_dir, branch=branch)
```

Comment:
can a osm github method be used instead? i have been trying to drop the git dependency

Reply:
I think I can refactor to use `download_url_from_github_url`